k8s/operators/src/pool/diskpool/client.rs

```diff
@@ -35,33 +64,149 @@ pub(crate) async fn create(
 /// is wrong would be to consult the logs.
 pub(crate) async fn ensure_crd(k8s: &Client) -> Result<CustomResourceDefinition, Error> {
     let crd_api: Api<CustomResourceDefinition> = Api::all(k8s.clone());
-
     // Check if there is an existing dsp diskpool. If yes, Replace it with new generated one.
     // Create new if it doesnt exist.
     let mut crd = DiskPool::crd();
     let crd_name = crd.metadata.name.as_ref().ok_or(Error::InvalidCRField {
         field: "diskpool.metadata.name".to_string(),
     })?;
+    crd.spec.versions[0].served = false;
+    let new_crd = V1beta1DiskPool::crd();
+    let all_crds = vec![crd.clone(), new_crd.clone()];
+    let new_crd = merge_crds(all_crds, "v1beta1").unwrap();
 
-    // Ensure the diskpool.
-    let result = if let Ok(existing) = crd_api.get(crd_name).await {
-        crd.metadata.resource_version = existing.resource_version();
+    // Ensure if diskpool crd exist to determine api verb.
+    let result = if crd_api.get(crd_name).await.is_ok() {
         info!(
             "Replacing CRD: {}",
-            serde_json::to_string_pretty(&crd).unwrap()
+            serde_json::to_string_pretty(&new_crd).unwrap()
         );
-
-        let pp = PostParams::default();
-        crd_api.replace(crd_name, &pp, &crd).await
+        let param = PatchParams::apply("merge_v1alpha1_v1beta1").force();
+        crd_api
+            .patch("diskpools.openebs.io", &param, &Patch::Apply(&new_crd))
+            .await
     } else {
         info!(
             "Creating CRD: {}",
-            serde_json::to_string_pretty(&crd).unwrap()
+            serde_json::to_string_pretty(&new_crd).unwrap()
         );
 
         let pp = PostParams::default();
-        crd_api.create(&pp, &crd).await
+        crd_api.create(&pp, &new_crd).await
     };
     let crd = result.map_err(|e| Error::Kube { source: e })?;
     Ok(crd)
 }
+
+/// This lists pool from control plane. Checks if DSP cr present for it. If no, create it ourself.
+pub(crate) async fn create_missing_cr(
```

Comment:
Can we add this as separate PR?

Reply:
It has to be after this gets merged then. Is that okay?

Reply:
Yeah I think so

Reply:
Okay